cloudflare-workers.js

```diff
@@ -0,0 +1,24 @@
+addEventListener('fetch', event => {
+    event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+    return (
+        (await handleSemanticAssetsFilenames(request)) || (await fetch(request))
+    )
+}
+
+async function handleSemanticAssetsFilenames(request) {
+    const re = /^https:\/\/assets.serlo.org\/(legacy\/|)(\w+)\/([\w\-\+]+)\.(\w+)$/
```

Comment:
This shouldn't match the current links, right? So e.g. https://assets.serlo.org/somehash.png is not matched. Thing is that https://assets.serlo.org/lecacy/somehash.org will be matched, with legacy matched by `(\w+)` group... not sure if thats a problem, though, can you check that?

Also why is the plus sign needed in the filname part `[\w\-\+]`?

Reply:
It definitely also works for legacy links, so you can also do https://assets.serlo.org/legacy/hash/name.ext. But you are right that old links w/o semantic names now get wrongly matched and don’t work correctly, probably. I’ll check that, good catch. And for the file name: I just added all allowed symbols that aren’t to far fetched (like $!?), we can probably live w/o +, though. We only need the hyphen for spaces I guess

Reply:
Ah ok, check. Hyphens is important, yes, I'm fine with or without the + character, so I'll leave that for you to decide.